### PR TITLE
Added a unit tests for pulling bazaar sources from URIs.

### PR DESCRIPTION
**Pull request by fazerlicourice7**
I created a test for bazaar type sources. The test ensures that sources.py correctly recognizes and calls the correct methods for bazaar sources.

LP: #1536283

### CODE REVIEW

**Comment by fazerlicourice7** [comment]
I'm trying to squash the merge commit, but when I attempt to rebase I can't see it as a commit.

File: debian/changelog [hunk 1]
```diff
@@ -86,6 +86,7 @@ snapcraft (2.0) xenial; urgency=medium
 
   [ Vamshi Balanaga ]
   * Added test for git sources in sources.py (LP: #1534411)
+  * Added test for bzr sources in sources.py (LP: #1536283)
```
Comment (elopio) [inline debian/changelog]: You shouldn't edit the changelog of the released version.
We could open a new section for the unreleased 2.1 version, but we are currently trying to do that automatically with git-buildpackage, so just remove this line, please.

**Comment by elopio** [comment]
The test looks good to me. I will let @kyrofa help you with the squash, because I might just make things worst.
Once you are able to rewrite the commit, please make sure to write it something like this:

Added a unit tests for pulling bazaar sources from URIs.

LP: #1536283

(note the capital letter at the start, and the period at the end. And it's not necessary to mention the file you changed but I consider it good to give a little more context of what you are doing)
We are taking care a lot about commit messages now because we are trying to autogenerate the changelog from them.

Thanks again @fazerlicourice7!

**Comment by kyrofa** [comment]
Looks good, though since the PR title turns into the merge commit message, let's leave it clean and leave off the bug ref there (just in the PR description and commit message is perfect).

**Comment by fazerlicourice7** [comment]
Thanks.

**Pull request merged**